src/Chip/Chip.js

```diff
@@ -0,0 +1,271 @@
+    onTouchEnd: () => {},
+    onTouchStart: () => {},
+  };
+
+  static contextTypes = {muiTheme: React.PropTypes.object.isRequired};
+
+  state = {clicked: false, focused: false};
+
+  handleBlur = (event) => {
+    this.setState({clicked: false, focused: false});
+    this.props.onBlur(event);
+  };
+
+  handleCloseIconTouchTap = (event) => {
+    // Stop the event from bubbling up to the `Chip`
+    event.stopPropagation();
+    this.props.onRequestClose(event);
+  };
+
+  handleFocus= (event) => {
+    this.setState({focused: true});
+    this.props.onFocus(event);
+  };
+
+  handleKeyboardFocus = (event, keyboardFocused) => {
+    if (keyboardFocused) {
+      this.handleFocus();
+    } else if (!this.state.hovered) {
+      this.handleBlur();
+    }
+
+    this.props.onKeyboardFocus(event, keyboardFocused);
+  };
+
+  handleKeyDown = (event) => {
+    if (keycode(event) === 'backspace') {
+      event.preventDefault();
+      if (this.props.onRequestClose) {
+        this.props.onRequestClose(event);
+      }
+    }
+    this.props.onKeyDown(event);
+  };
+
+  handleMouseDown = (event) => {
+    // Only listen to left clicks
+    if (event.button === 0) {
+      event.stopPropagation();
+      this.setState({clicked: true});
+    }
+    this.props.onMouseDown(event);
+  };
+
+  handleMouseEnter = (event) => {
+    this.handleFocus();
+    this.props.onMouseEnter(event);
+  };
+
+  handleMouseLeave = (event) => {
+    if (!this.refs.button.isKeyboardFocused()) {
+      this.handleBlur();
+    }
+    this.props.onMouseLeave(event);
+  };
+
+  handleMouseUp = (event) => {
+    this.setState({clicked: false});
+    this.props.onMouseUp(event);
+  };
+
+  handleTouchEnd = (event) => {
+    this.setState({clicked: false});
+    this.props.onTouchEnd(event);
+  };
+
+  handleTouchStart = (event) => {
+    event.stopPropagation();
+    this.setState({clicked: true});
+    this.props.onTouchStart(event);
+  };
+
+
+  render() {
+    const buttonEventHandlers = {
+      onBlur: this.handleBlur,
+      onFocus: this.handleFocus,
+      onKeyDown: this.handleKeyDown,
+      onMouseDown: this.handleMouseDown,
+      onMouseEnter: this.handleMouseEnter,
+      onMouseLeave: this.handleMouseLeave,
+      onMouseUp: this.handleMouseUp,
+      onTouchEnd: this.handleTouchEnd,
+      onTouchStart: this.handleTouchStart,
+      onKeyboardFocus: this.handleKeyboardFocus,
+    };
+
+    const {prepareStyles} = this.context.muiTheme;
+    const styles = getStyles(this.props, this.context, this.state);
+
+    let {children, style, className, labelStyle, ...other} = this.props;
+    const deletable = this.props.onRequestClose;
+    let avatar = null;
+
+    style = Object.assign(styles.root, style);
+    labelStyle = prepareStyles(Object.assign(styles.label, labelStyle));
+
+    const deleteIcon = deletable ?
+      <DeleteIcon
+        color={styles.closeIcon.color}
+        style={styles.closeIcon}
+        onClick={this.handleCloseIconTouchTap}
```

Comment:
@oliviertassinari It's getting removed, that was my mistake -- It's also causing a bug at the moment in the docs 👍